### ISSUE TITLE
Add option to normalize URLs and remove duplicate slashes

### ISSUE THREAD

**Issue by shadyabhi** [apache/trafficserver#9609]
There are a lot of other URL normalizations a proxy can support, but personally, for our use-case, we're only interested in merging duplicate slashes. 

To give more clarity on what's required, here are docs from HAproxy around this. 

HAproxy: https://docs.haproxy.org/dev/configuration.html#http-request%20normalize-uri%20path-merge-slashes
Implementation: https://github.com/haproxy/haproxy/commit/d371e99d1c64744b906a5c9970b8a7d35bf08ae8

Original URL: https://domain.com/abc//def
Normalized URL before fetching from origin: https://domain.com/abc/def

**Comment by bryancall** [comment]
This sounds reasonable.  We already git rid of one of the slashes.

**Comment by shadyabhi** [comment]
>We already git rid of one of the slashes.

Thank you for response @bryancall, to confirm, are you saying that ATS already normalizes duplicate slashes? I wasn't able to find any references in docs.

**Comment by shadyabhi** [comment]
>This sounds reasonable. We already git rid of one of the slashes.

@bryancall Can you help reference the location where this is done and then we can close this ticket with that reference? Thanks!

**Comment by bryancall** [comment]
This issue was fixed in CVE-2023-33934 (https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-33934).  There was a paper on differential fuzzing (https://bahruz.me/publications/35108) that showed it was easy to bypass remap rules using multiple slashes at the beginning of the URL.  We determined it was better not to remove extra slashes in other parts of the path, since they are considered valid URLs.
